.githooks/commit-msg

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+import fnmatch
+import os.path
+import sys
+import subprocess
+
+CI_IGNORE_FILE='.circleci/ciignore'
+CI_IGNORE_TAG='[skip ci]'
+GIT_CMD=['git', 'diff', '--cached', '--name-only']
```

Comment:
Perhaps this should compare to what's in master?

Wouldn't you be able to do the following otherwise:

1) Commit change to C++ code that fails tests with CI failing.
2) Commit change to readme, CI is now skipped and build becomes green?

Reply:
The idea here is to skip CI for intermediate commits when working on a PR. eg: you make code changes, push them up, then in a separate commit push just a changelog change. This will prevent running CI for just the changelog change. Also, if we make it ask the user, then they can decide if they want to skip CI or not.

If we diff with master, then we'll almost always run CI, since a PR changing just non-buildable files is quite rare, but changing them in a series of commit that make a PR might be more common.

Whats not very clear from the docs is what happens to in-progress runs when a push with `skip ci` is made. Say I do a code push, which triggers CI, and then before that run completes, I push a skippable change. Looks like the previous run still runs, but I can't see its progress in github UI. I need to do a bit more testing to see how it behaves.

Reply:
Okay, so as per my tests, if a previous CI run is in progress and you push a commit with `[skip ci]` (or `[ci skip]`), the existing CI run continues on (and you can see its status against the commit sha) BUT, circle CI does not invoke a completion callback to github. And it seems like this prevents from the merge button from being activated (even though the previous run was successful) 

This makes using `skip ci` not very useful (it will work if we let the last CI run run fully and update status, before pushing the skip CI commit, but then thats hardly useful). :disappointed:

Let me see if there's a way to use parameters and use conditionals within CI config to skip CI runs (i.e circle jobs still run, but they dont execute certain workflows). This should keep github happy as the circle callback would still get invoked.

Reply:
Doh, the merge button is disabled because the PR is not approved :man_facepalming: Although the rest of my issues with skip-ci still remain.

Reply:
Looks like I was right after all, the status needs to be passing for the merge button to enable:
![Screenshot from 2021-04-09 17-39-17](https://user-images.githubusercontent.com/4003000/114252553-d2b07300-9973-11eb-847b-7578d97d6494.png)

Back to square 1 :/